multiapps-controller-core/src/main/java/org/cloudfoundry/multiapps/controller/core/util/NameUtil.java

```diff
@@ -73,19 +59,62 @@ public static String getNameWithProperLength(String name, int maxLength) {
         return name;
     }
 
-    public static String computeNamespacedNameWithLength(String name, String namespace, boolean applyNamespace, int maxLength) {
-        String prefix = "";
+    public static String computeNamespacedNameWithLength(String name, String namespace, boolean applyNamespace,
+                                                         boolean applyNamespaceAsSuffix, int maxLength) {
         if (StringUtils.isNotEmpty(namespace) && applyNamespace) {
-            prefix = getNamespacePrefix(namespace);
+            if (applyNamespaceAsSuffix) {
+                name = getNameWithNamespaceSuffix(name, namespace, maxLength);
+            } else {
+                name = getNamespacePrefix(namespace) + name;
+            }
         }
+        return getNameWithProperLength(name, maxLength);
+    }
 
-        return getNameWithProperLength(prefix + name, maxLength);
+    private static String getNameWithNamespaceSuffix(String name, String namespace, int maxLength) {
+        String namespaceSuffix = getNamespaceSuffix(namespace);
+        String nameWithoutShortening = name;
+        name = getNameWithProperLength(name, calculateNameLengthWithoutNamespaceAndBlueGreenSuffix(namespaceSuffix, maxLength));
```

Comment:
you can skip the above line by simply adding
String shortenedName = getNameWithProperLength(..) and don't need to change the value of 'name'